pkg/virt-launcher/notify-client/client.go

```diff
@@ -136,8 +137,11 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, libvirtEvent lib
 		client.SendDomainEvent(watchEvent)
 		events <- watchEvent
 	default:
-		if libvirtEvent.Event != nil {
-			if libvirtEvent.Event.Event == libvirt.DOMAIN_EVENT_DEFINED && libvirt.DomainEventDefinedDetailType(libvirtEvent.Event.Detail) == libvirt.DOMAIN_EVENT_DEFINED_ADDED {
+		if libvirtEvent.Event != nil || interfaceStatus != nil {
+			if interfaceStatus != nil {
```

Comment:
We don't have libvirtEvent.Event, so we can not use it to do the check in the following if's.
But whenever we have the interface statuses, we want to update, as we know the agent is running, and the state of the interfaces changed (agent poller is checking that)

Reply:
that's one of the reasons why I want the callback being invoked in a central place, where we always have the latest data, so that we simply assemble the final picture in the callback.

Reply:
The callback is now done from one place.
I'm not quite sure if this is what you asked about:
We send the domain event (SendDomainEvent) either in response to libvirt domain event, or to an agent event, and the conditions for deciding to send the domain event are different in either case (hence the if's).
We could modify eventCallback to take agentUpdateEvent instead of InterfaceStatus if that will be more clear.